assembly/src/tests.rs

```diff
@@ -1313,6 +1314,45 @@ fn const_conversion_failed_to_u32() -> TestResult {
     Ok(())
 }
 
+#[test]
+fn const_word_from_string() -> TestResult {
+    let context = TestContext::default();
+    let sample_source_string = "lorem ipsum";
+
+    let source = source_file!(
+        &context,
+        format!(
+            r#"
+    const.SAMPLE_WORD=to_word("{sample_source_string}")
```

Comment:
We are using `to_word` here because making `word` a keyword causes issues as it collides with the `word` module in the standard library. I don't really love having `to_word` be a keyword either - so, curious what could be the alternatives.

Maybe we could use some special notation instead - e.g., something like:
```
const.MY_WORD=<"my_string">
```
But I'm not sure i like this approach either. @bitwalker, @adr1anh, @PhilippGackstatter - what do you think?

Reply:
I don't understand the bit about this colliding with the `word` module - it sounds to me like the issue is that the `word` token was added without also adding it as a valid identifier token (in a different part of the grammar, and is needed for any token which is also a valid identifier).

Reply:
Ah, now I see, thank you!